rclcpp_examples/src/services/add_two_ints_client.cpp

```diff
@@ -35,6 +36,47 @@ example_interfaces::srv::AddTwoInts_Response::SharedPtr send_request(
   }
 }
 
+// This function could be considered a prototype of an API that should be
+// promoted into rclcpp. This version polls; doing it in an event-driven
+// manner would be better.
+bool wait_for_service(rclcpp::Node::SharedPtr node,
+  const std::string & topic,
+  const std::chrono::steady_clock::duration timeout)
+{
+  const std::chrono::milliseconds sleep_per_loop(1);
+  auto end = std::chrono::steady_clock::now() + timeout;
+  bool success = false;
+  bool printed = false;
+  while (std::chrono::steady_clock::now() < end) {
+    try {
+      if (node->count_subscribers(topic) == 0) {
+        success = true;
+        break;
+      }
+    } catch (std::runtime_error & e) {
```

Comment:
The underlying function `rmw_count_subscribers()` is C (for fastrtps, it's hardcoded to return an error: https://github.com/eProsima/ROS-RMW-Fast-RTPS-cpp/blob/master/rmw_fastrtps_cpp/src/functions.cpp#L1793-L1801), but here I'm calling a C++ class method: https://github.com/ros2/rclcpp/blob/master/rclcpp/src/rclcpp/node.cpp#L317-L329.